tools/go-cli/go-whisk-cli/commands/action.go

```diff
@@ -216,30 +216,36 @@ var actionInvokeCmd = &cobra.Command{
             payload = (*json.RawMessage)(&data)
         }
 
+        outputStream := os.Stdout
+
         activation, _, err := client.Actions.Invoke(qName.entityName, payload, flags.common.blocking)
         if err != nil {
-            whisk.Debug(whisk.DbgError, "client.Actions.Invoke(%s, %s, %t) error: %s\n", qName.entityName, payload, flags.common.blocking, err)
-            errMsg := fmt.Sprintf("Unable to invoke action '%s': %s", qName.entityName, err)
-            whiskErr := whisk.MakeWskErrorFromWskError(errors.New(errMsg), err, whisk.EXITCODE_ERR_GENERAL,
-                whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
-            return whiskErr
+            whiskErr, isWhiskErr := err.(*whisk.WskError)
+
+            if (isWhiskErr && whiskErr.ApplicationError != true) || !isWhiskErr {
+                whisk.Debug(whisk.DbgError, "client.Actions.Invoke(%s, %s, %t) error: %s\n", qName.entityName, payload,
+                    flags.common.blocking, err)
+                errMsg := fmt.Sprintf("Unable to invoke action '%s': %s", qName.entityName, err)
```

Comment:
This error message shouldn't say "unable to invoke action" - that at least to me implied there was an internal error and the system was unable to handle the request. But for an application error, the system did in fact invoke the action, it's just that the action failed/was buggy.